app/src/main/java/com/github/polybooks/FilteringBooksActivity.kt

```diff
@@ -0,0 +1,145 @@
+    private lateinit var mSemBa3 : CheckBox
+    private lateinit var mSemMa1 : CheckBox
+    private lateinit var mSemMa2 : CheckBox
+
+    private lateinit var mCourseCS306 : CheckBox
+    private lateinit var mCourseCOM480 : CheckBox
+
+    //------
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_filtering_books)
+
+        // Get a reference to the UI parameters
+        mResetParameters = findViewById(R.id.reset_button)
+        mResults = findViewById(R.id.results_button)
+
+        // Set behaviour Reset and Results
+        setResetButtonBehaviour()
+        setResultsButtonBehaviour()
+
+        // --- TODO hardcoded : make it dynamic
+        setParametersButtons()
+    }
+
+    private fun setResetButtonBehaviour() {
+        mResetParameters.setOnClickListener {
+
+            mSortGroup.clearCheck()
+
+            mFieldCS.isChecked = false
+            mFieldBio.isChecked = false
+            mFieldArchi.isChecked = false
+            mSemBa1.isChecked = false
+            mSemBa2.isChecked = false
+            mSemBa3.isChecked = false
+            mSemMa1.isChecked = false
+            mSemMa2.isChecked = false
+            mCourseCS306.isChecked = false
+            mCourseCOM480.isChecked = false
+        }
+    }
+
+    private fun setResultsButtonBehaviour() {
+        mResults.setOnClickListener {
+
+            var query : BookQuery = DummyBookQuery()
+                    .onlyIncludeInterests(getInterests())
+                    .withOrdering(getOrdering())
+
+            val intent : Intent = Intent(this, ListSalesActivity::class.java)
+            intent.putExtra(ListSalesActivity.EXTRA_BOOKS_QUERY, query)
+            startActivity(intent)
+        }
+    }
+
+    private fun setParametersButtons() {
+        mSortGroup = findViewById(R.id.sort_group)
+        mSortPopularity = findViewById(R.id.popularity_sort)
+        mSortTitleInc = findViewById(R.id.title_inc_sort)
+        mSortTitleDec = findViewById(R.id.title_dec_sort)
+
+        mFieldCS = findViewById(R.id.CS)
+        mFieldBio = findViewById(R.id.Biology)
+        mFieldArchi = findViewById(R.id.Archi)
+
+        mSemBa1 = findViewById(R.id.ic_ba1)
+        mSemBa2 = findViewById(R.id.ma_ba2)
+        mSemBa3 = findViewById(R.id.sv_ba3)
+        mSemMa1 = findViewById(R.id.gc_ma1)
+        mSemMa2 = findViewById(R.id.mt_ma2)
+
+        mCourseCS306 = findViewById(R.id.CS306)
+        mCourseCOM480 = findViewById(R.id.COM480)
+    }
+
+    private fun getOrdering() : BookOrdering {
+        if(mSortPopularity.isChecked) return BookOrdering.DEFAULT //add it when available
+        if(mSortTitleDec.isChecked) return BookOrdering.TITLE_DEC //add it when available
+        if(mSortTitleInc.isChecked) return BookOrdering.TITLE_INC //add it when available
+        else return BookOrdering.DEFAULT
+    }
+
+    private fun getInterests() : Set<Interest> {
```

Comment:
Should get adapted to the Spinner (and that's kinda the tricky part of using the Spinner I think)

Reply:
See above